plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteTypeMapping.java

```diff
@@ -0,0 +1,387 @@
+
+        binaryTest("varbinary")
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_varbinary"))
+                .execute(getQueryRunner(), igniteCreateAndInsert("test_varbinary"));
+    }
+
+    private static SqlDataTypeTest binaryTest(String inputType)
+    {
+        return SqlDataTypeTest.create()
+                .addRoundTrip(inputType, "X'68656C6C6F'", VARBINARY, "to_utf8('hello')")
+                .addRoundTrip(inputType, "X''", VARBINARY, "X''")
+                .addRoundTrip(inputType, "NULL", VARBINARY, "CAST(NULL AS varbinary)")
+                .addRoundTrip(inputType, "X'5069C4996B6E6120C582C4856B61207720E69DB1E4BAACE983BD'", VARBINARY, "to_utf8('Piękna łąka w 東京都')")
+                .addRoundTrip(inputType, "X'4261672066756C6C206F6620F09F92B0'", VARBINARY, "to_utf8('Bag full of 💰')")
+                .addRoundTrip(inputType, "X'0001020304050607080DF9367AA7000000'", VARBINARY, "X'0001020304050607080DF9367AA7000000'") // non-text
+                .addRoundTrip(inputType, "X'000000000000'", VARBINARY, "X'000000000000'");
+    }
+
+    @Test(dataProvider = "sessionZonesDataProvider")
+    public void testDate(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                .build();
+        SqlDataTypeTest.create()
+                .addRoundTrip("int", "1", INTEGER, "1") // Avoid using date as primary key in Ignite, https://issues.apache.org/jira/browse/IGNITE-8552
+                .addRoundTrip("date", "DATE '1970-01-01'", DATE, "DATE '1970-01-01'") // change forward at midnight in JVM
+                .addRoundTrip("date", "DATE '1970-02-03'", DATE, "DATE '1970-02-03'")
+                .addRoundTrip("date", "DATE '2017-07-01'", DATE, "DATE '2017-07-01'") // summer on northern hemisphere (possible DST)
+                .addRoundTrip("date", "DATE '2017-01-01'", DATE, "DATE '2017-01-01'") // winter on northern hemisphere (possible DST on southern hemisphere)
+                .addRoundTrip("date", "DATE '1983-04-01'", DATE, "DATE '1983-04-01'") // change forward at midnight in Vilnius
+                .addRoundTrip("date", "DATE '1983-10-01'", DATE, "DATE '1983-10-01'")
+                .addRoundTrip("date", "DATE '2983-10-01'", DATE, "DATE '2983-10-01'")
+                .addRoundTrip("date", "DATE '3983-10-01'", DATE, "DATE '3983-10-01'")
+                .addRoundTrip("date", "DATE '4983-10-01'", DATE, "DATE '4983-10-01'")
+                .addRoundTrip("date", "DATE '9983-10-01'", DATE, "DATE '9983-10-01'")
+                .addRoundTrip("date", "DATE '9999-01-01'", DATE, "DATE '9999-01-01'")
+                .addRoundTrip("date", "DATE '9999-12-31'", DATE, "DATE '9999-12-31'")
+                .execute(getQueryRunner(), session, trinoCreateAndInsert("test_date"))
+                .execute(getQueryRunner(), session, igniteCreateAndInsert("test_date"));
+    }
+
```

Comment:
Please add another test method to verify the failure about out of supported date ranges.